plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -130,6 +130,7 @@ export class KubernetesProxy {
   // single middleware instance - and we don't expect it to change over time.
   private async getMiddleware(originalReq: Request): Promise<RequestHandler> {
     const originalCluster = await this.getClusterForRequest(originalReq);
+    const originUrl = new URL(originalCluster.url);
```

Comment:
This value will be persisted in memory as a property of `middleware` the first time a request is proxied to this cluster...